app/pages/Pdf/ServicePdfPage.tsx

```diff
@@ -0,0 +1,421 @@
+    [service]
+  );
+
+  const ref = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    if (id) {
+      fetchService(id).then((s) => {
+        setService(s);
+      });
+    }
+  }, [id]);
+
+  useEffect(() => {
+    if (mapImgSrc !== null && !pdfSource) {
+      // Map image has been fetched. We're now ready to make our request to the PDF conversion endpoint
+      fetch(`/api/services/html_to_pdf`, {
+        body: JSON.stringify({ html: ref.current?.outerHTML }),
+        method: "POST",
+        mode: "cors",
+        headers: { "Content-Type": "application/json" },
+      })
+        .then((resp) => resp.blob())
+        .then((blob) => {
+          setPdfSource(window.URL.createObjectURL(blob));
+        });
+    }
+  }, [mapImgSrc, pdfSource]);
+
+  if (!service) {
+    return <Loader />;
+  }
+
+  const { resource, recurringSchedule } = service;
+  const locations = getServiceLocations(service, resource, recurringSchedule);
+
+  if (locations.length > 0 && mapImgSrc === null) {
+    const { address_1, city, state_province, postal_code } =
+      locations[0].address;
+
+    // Use address rather than Lat/Lng as center. In some cases, a location may not have
+    // lat/lng or it may not be updated after address changes. Address seems to
+    // be a safer value to access.
+    const center = `${address_1}, ${city}, ${state_province}, ${postal_code}`;
+    const baseMapUrl = "https://maps.googleapis.com/maps/api/staticmap";
+    const params = {
+      center,
+      zoom: "14",
+      size: "375x225",
+      markers: `color:0xff6821|${center}`,
+      maptype: "roadmap",
+      key: config.GOOGLE_API_KEY,
+    };
+
+    fetch(
+      `${baseMapUrl}?${qs.stringify(params, { encodeValuesOnly: true })}`
+    ).then((mapResp) => {
+      setMapImgSrc(mapResp.url);
+    }).catch(() => {
+      // If there is an error, set the mapImgSrc to non-null so that the HTML to PDF API request will still be made
+      setMapImgSrc('');
+    });
+  }
+
+  return (
+    <div>
+      {pdfSource && (
+        <embed
+          src={pdfSource}
+          type="application/pdf"
+          height="100%"
+          width="100%"
+          className={styles.pdfEmbed}
+        />
+      )}
+      {!pdfSource && (
+        <>
+          <p className={styles.loaderText}>Fetching PDF...</p>
+          <Loader />
+        </>
+      )}
+      ;
+      {!pdfSource && (
+        <div
+          ref={ref}
+          className={`serviceHandoutPdf ${styles.handoutInvisible}`}
+        >
+          {/* The below styles contain some overrides of global .renderedMarkdown and table tag selector styles */}
+          <style>
+            {`
+          .serviceHandoutPdf * {
+            margin: 0;
+            font-family: "Open Sans", "San Francisco", "Roboto", "Arial", sans-serif;
+          }
+
+          .serviceHandoutPdf {
+            padding: 0 9px 9px;
+            background-color: #fff;
+          }
+
+          .serviceHandoutPdf table.compact tr th,
```

Comment:
Also, I think I might be mistaking some of the table elements that come from our other components vs. the ones that are actually directly in the React code below. For the ones coming from the other components, is it at all possible to use CSS modules to inline the generated CSS class names into this one? I'm not 100% certain, but I think that you can just import the stylesheets for those other components into this file and then just render the individual properties into this inline stylesheet, since the imported stylesheet as a JavaScript object should have properties that resolve to the mangled CSS class names.

Reply:
See my comment below. I went ahead copied over the external components into this file. I added some classes to the copied components but not all in the interest of time. The components that this file was referencing (and now that have been copied over) were not using CSS modules; they using a global stylesheet. Anyhow, see my comment below for more details. Thanks!